third_party/blink/renderer/core/dom/element.cc

```diff
@@ -642,7 +643,11 @@ bool IsGuaranteedToEnterNGBlockNodeLayout(const LayoutObject& layout_object) {
 Element::Element(const QualifiedName& tag_name,
                  Document* document,
                  ConstructionType type)
-    : ContainerNode(document, type), tag_name_(tag_name) {}
+    : ContainerNode(document, type), tag_name_(tag_name) {
+  // RUN-1304 - Elements can be constructed non-deterministically,
+  // but we don't understand where.  This should help us get to the bottom of that.
+  recordreplay::Assert("[RUN-1304-1359] Element::Element %08x", (unsigned) type);
```

Comment:
I still am unsure: Since the problem occurs when `AreEventsDisabled()`, does this `Assert` actually generate a complete mismatch, or will it just... not give us the info that we so desire? I'd have to defer to @bhackett1024 on this one 😞

Reply:
What I expect this assertion to do is be hit when events are disallowed.  It won't generate a Mismatch or a CallStackMismatch, but it should generate a RecordingWarning with "Assertion during events disallowed" message, with the relevant message embedded.  That should tell us when `Element` instances are being constructed in non-deterministic contexts.